source/math/FloatExtension.ooc

```diff
@@ -110,7 +110,7 @@ extend Float {
 				value /= 10.0f
 				radix *= 10.0f
 			}
-			while (Float absolute(value) < pow(10.0f, valueDigits-1)) {
+			while (Float absolute(value) - pow(10.0f, valueDigits-1) < -0.000001) {
```

Comment:
Ah, missed that line. 1.0E-6 was chosen to achieve good precision and at the same time prevent truncation errors.